framework/src/actions/SetupMeshAction.C

```diff
@@ -236,10 +236,10 @@ SetupMeshAction::act()
           _type = "MeshGeneratorMesh";
           auto original_params = _moose_object_pars;
           _moose_object_pars = _factory.getValidParams("MeshGeneratorMesh");
-          std::vector<std::string> params_to_apply{"parallel_type", "patch_size", "uniform_refine"};
```

Comment:
This line of code is asking for trouble. I think it's best to just extract all of the parameters for the new object if it's programmatically changed.